dev/PushNotifications/PushNotificationChannel.cpp

```diff
@@ -21,21 +26,42 @@ namespace winrt::Microsoft
 
 namespace winrt::Microsoft::Windows::PushNotifications::implementation
 {
-    PushNotificationChannel::PushNotificationChannel(winrt::Windows::PushNotificationChannel const& channel): m_channel(channel) {}
-
     winrt::Windows::Uri PushNotificationChannel::Uri()
     {
-        return winrt::Windows::Uri{ m_channel.Uri() };
+        if (m_channel)
+        {
+            return winrt::Windows::Uri{ m_channel.Uri() };
+        }
+        else
+        {
+            return winrt::Windows::Uri{ m_channelInfo.channelUri };
+        }
     }
+
     winrt::Windows::DateTime PushNotificationChannel::ExpirationTime()
     {
-        return m_channel.ExpirationTime();
+        if (m_channel)
+        {
+            return m_channel.ExpirationTime();
+        }
+        else
+        {
+            return m_channelInfo.channelExpiryTime;
+        }
     }
+
     void PushNotificationChannel::Close()
     {
         try
         {
-            m_channel.Close();
+            if (m_channel)
+            {
+                m_channel.Close();
+            }
+            else
+            {
+               PushNotifications_CloseChannel(m_channelInfo.appUserModelId.c_str(), m_channelInfo.channelId.c_str());
+            }
 
             PushNotificationTelemetry::ChannelClosedByApi(S_OK);
```

Comment:
Unrelated but we should look for a bug fix here. We should really be checking if the API's above return an error instead of logging S_OK all the time. @loneursid

Reply:
Another important missing datapoint is the appId which we have to log with every event for these to be useful.

Reply:
we are loging the AppUserModelId. Isn't this enough

Reply:
It needs to log the processName in the case of unpackaged apps, I don't think we are doing it right now.

Reply:
PushNotificationTelemetry::ChannelClosedByApi(S_OK)

I thought this is just logging HRESULT. Not true?

Reply:
@danielayala94 @loneursid  I think for packaged apps, we can log the appUsermodelId. For unpackaged, we can simply log the process name. 

For the process name, please lets make sure that we remove the file path since it can contain PII data

Reply:
Telemetry was created before we had unpackaged app support. That's a task for after the feature has been implemented.

Reply:
The provider automatically add the AppUserModelId to all telemetry events.

Reply:
I don't see how thats possible for Unpackaged apps with no identity. For packaged apps, sure

Reply:
Unless the core telemetry logic is using processname somehow if the app is unpackaged?

Reply:
Most likely AppUserModelId is just null for unpackaged.

Reply:
Taking this as part of this bug to be addressed next week -> #1372